src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,99 @@
+    }
+
+    /**
+     * @param input The String input from the user.
+     * @return Whether "bye" has been inputted.
+     */
+    public boolean parse(String input) {
+        if (input.equals("bye")) {
+            return false;
+        } else if (input.equals("list")) {
+            list.printTaskList();
+        } else {
+            String[] split = input.split(" ", 2);
+            try {
+                if (split.length > 0 && Arrays.asList(validInputs).contains(split[0])) {
+                    switch (split[0]) {
+                    case "delete": // Checks for delete
+                        int index = Integer.parseInt(split[1]) - 1;
+                        this.list.deleteTask(index);
+                        break;
+                    case "mark": // Checks for mark
+                        index = Integer.parseInt(split[1]) - 1;
+                        this.list.markTask(index);
+                        break;
+                    case "unmark": // Checks for unmark
+                        index = Integer.parseInt(split[1]) - 1;
+                        this.list.unmarkTask(index);
+                        break;
+                    case "todo": // Checks for Todo
+                        if (split.length < 2) {
+                            throw new DukeException("todo");
+                        } else {
+                            String description = split[1];
+                            this.list.addTodo(description);
+                        }
+                        break;
+                    case "deadline": // Checks for Deadline
+                        if (split.length < 2) {
+                            throw new DukeException("deadline");
+                        } else {
+                            String[] temp = split[1].split(" /by ", 2);
+                            String description = temp[0];
+                            String by = temp[1];
+                            this.list.addDeadline(description, by);
+                        }
+                        break;
+                    case "event": // Checks for Event
+                        if (split.length < 2) {
+                            throw new DukeException("event");
+                        } else {
+                            String[] temp = split[1].split(" /at ", 2);
+                            String description = temp[0];
+                            String when = temp[1];
+                            this.list.addEvent(description, when);
+                        }
+                        break;
+                    case "find":
+                        if (split.length < 2) {
+                            throw new DukeException("find");
+                        } else {
+                            String keyword = split[1];
+                            this.list.findTask(keyword).printTaskList();
+                        }
+                        break;
+                    default: // Default case
```

Comment:
This is based on what I understand so far from your code.

You have a set of valid inputs declared earlier, and here you use the switch case statement to check which valid command the input matches, but this is only after an if statement basically checking if the first word matches any valid commands.

If so, then logically (assuming your program works as it should) the default case should never occur right? 

If that is so, I think it would make more sense to simply have it be empty, and instead, at **line 91** below (reached after there is an invalid input), the error message can be included as `"Wrong input"`, instead of being blank. Would that be better?